### DISCUSSION TITLE
MongoParseError: URI malformed, cannot be parsed

### DISCUSSION THREAD

**Discussion by Nonzzo** [trendscenter/coinstac#1458] [Q&A] [answered]
I am setting up the coinstac for development in a mac (I believe this is not an issue in a windows environment) and having the following error: "MongoParseError: URI malformed, cannot be parsed".  Anyone having the same issues? I have created a .env file and populated it with the environmental variables in the config/.env-example.sh contents.

**Comment by rssk** [comment]
So that seems like an issue with your mongo conn string envar which is set to this in the env 
`MONGODB_CONN_STRING=mongodb://root:test@localhost:27017`
This is consumed here
https://github.com/trendscenter/coinstac/blob/master/packages/coinstac-api-server/src/database/index.js#L8
I'd take a look and make sure that value is getting to there, the error you're getting isnt an auth error but rather the conn string itself isnt how mongo is expecting it

**Reply by Nonzzo** [reply]
Using the shell variables instead of environmental variables works. Thanks!

**Comment by dylanmartin** [comment, marked as answer]
@Nonzzo you don't set environment variables by creating a .env file, you need to set them in the shell you are using. If you are on Mac, the file you should modify is likely `.zshrc`
https://linuxhint.com/set-environment-variable-zsh/
once you set those variables, you should be able to enter your shell and type `env` to see that the key/values are set.

**Reply by Nonzzo** [reply]
Thanks It works!

**Reply by Nonzzo** [reply]
Also for mac .bashrc can work. You can make sure by checking the shell you rae using the command: echo $SHELL
if you get /bin/bash you can use .bashrc. Anyways the most important thing is to activate whatever shell you are using for the task with the source ~/.zshrc or ~/.bashrc depending on what you are using.
Also confirm your env variable by using the command: echo $env_variable (where env_variable could be any of the env variable), if it returns the value you are good to go.
